packages/flutter/lib/src/services/image_provider.dart

```diff
@@ -296,15 +297,36 @@ class NetworkImage extends DataPipeImageProvider<NetworkImage> {
   }
 
   @override
-  Future<mojo.MojoDataPipeConsumer> loadDataPipe(NetworkImage key) async {
-    assert(key == this);
-    return http.readDataPipe(Uri.base.resolve(key.url));
+  ImageStreamCompleter load(NetworkImage key) {
+    return new OneFrameImageStreamCompleter(
+      _loadAsync(key),
+      informationCollector: (StringBuffer information) {
+        information.writeln('Image provider: $this');
+        information.write('Image key: $key');
+      }
+    );
   }
 
-  @override
-  double getScale(NetworkImage key) {
+  Future<ImageInfo> _loadAsync(NetworkImage key) async {
     assert(key == this);
-    return key.scale;
+
+    final Uri resolved = Uri.base.resolve(key.url);
+    final http.Response response = await http.get(resolved);
+    if (response == null || response.statusCode != 200)
+      throw 'Unable to read image data from url.';
```

Comment:
can you throw FlutterError objects instead? We're planning on turning on the lint that whines about non-Error exception objects (like strings).